### ISSUE TITLE
Eager Materialization only executes 1 single downstream partition even though downstream asset contains multiple dependent partitions

### ISSUE THREAD

**Issue by someaveragepunter** [dagster-io/dagster#15383]
### Dagster version

1.3.9

### What's the issue?

this can happen when the downstream asset is MultiPartitioned or Partitions mapped to multiple cross product partitions from multiple upstream assets of different partitions.

### What did you expect to happen?

I'd expect once **b1** is materialized, all 3 of the downstream [b1.c1, b1.c2, b1.c3] to auto materialize as well. unfortunately only 1 of them materializes. there seems to be a 1-1 partition correspondence btw upstream and downstream partitions despite using MultiPartitions or StaticPartitionMappings

### How to reproduce?

```
import itertools
from dagster import (
    AssetIn,
    MultiPartitionsDefinition,
    StaticPartitionMapping,
    StaticPartitionsDefinition,
    asset, AutoMaterializePolicy,
)
A = StaticPartitionsDefinition(["a1", "a2", "a3"])
B = StaticPartitionsDefinition(["b1", "b2", "b3"])
C = StaticPartitionsDefinition(["c1", "c2", "c3"])
BC = StaticPartitionsDefinition(
    # instead of actual multi-partitions, just create strings of the form b1.c1
    [f"{b}.{c}" for b, c in itertools.product(B.get_partition_keys(), C.get_partition_keys())]
)
ABC = MultiPartitionsDefinition(
    {
        "a": A,
        "bc": BC,
    }
)
@asset(partitions_def=A)
def assetA():
    return [1,2,3]
@asset(partitions_def=B)
def assetB():
    return ['a', 'b']
@asset(partitions_def=C)
def assetC():
    return [11, 22, 33]
@asset(
    partitions_def=BC,
    auto_materialize_policy=AutoMaterializePolicy.eager(),
    ins={
        "assetB": AssetIn(
            partition_mapping=StaticPartitionMapping(
                {
                    # each partition of b maps to...
                    b_partition: {
                        # all multi partition keys that contain that partition
                        f"{b_partition}.{c_partition}"
                        for c_partition in C.get_partition_keys()
                    }
                    for b_partition in B.get_partition_keys()
                }
            ),
        ),
        "assetC": AssetIn(
            partition_mapping=StaticPartitionMapping(
                {
                    # each partition of c maps to...
                    c_partition: {
                        # all multi partition keys that contain that partition
                        f"{b_partition}.{c_partition}"
                        for b_partition in B.get_partition_keys()
                    }
                    for c_partition in C.get_partition_keys()
                }
            ),
        ),
    },
)
def assetBC(assetB, assetC):
    return assetB + assetC
@asset(partitions_def=ABC,
       auto_materialize_policy=AutoMaterializePolicy.eager(),
       )
def assetABC(assetA, assetBC):
    return assetA + assetBC





```

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by OwenKephart** [comment]
Hi @someaveragepunter ! I think this is due to the default `max_materializations_per_minute` value of 1 on eager auto_materialize_policies. If you want all partitions of an asset to be materialized in a single tick, then you can use `AutoMaterializePolicy.eager(max_materializations_per_minute=None)`, or, if you want some sort of guardrails, something like `AutoMaterializePolicy.eager(max_materializations_per_minute=<some high number>)`.

I would caution though that depending on the dimensionality of your static partitions this might have some performance implications (e.g. launching 100s or thousands of runs in response to a single event might not be ideal)

**Comment by daddywantssugar** [comment]
thanks, that seems to behave correctly or Non-Multipartitions like my example above, however when I use **MultiPartitions** like in this example I'd sent you in separate post below, dagster only materializes across one staticpartition but only for a single date (latest in this case)   

https://gist.github.com/OwenKephart/600e1c791f192ea78ef06ac8dc944e42?permalink_comment_id=4634120#gistcomment-4634120

P.S. changing title of this bug report to reflect this case

**Comment by daddywantssugar** [comment]
Update
this constraint has now been resolved in 1.3.14 thank you @OwenKephart
